### DISCUSSION TITLE
Handle this exception Query affected 3 rows while 1 expected

### DISCUSSION THREAD

**Discussion by zbouchlih** [serenity-is/Serenity#6123] [Help] [answered]
Hey All,

I have a many to many relationship (**type_contract**) and I want to manage it in **TypeForm**

I have this field in **TypeRow**
:
        [DisplayName("Contracts")]
        [LookupEditor(typeof(ContractRow), Multiple = true, FilterField = "Actif", FilterValue = true), NotMapped]
        [LinkingSetRelation(typeof(ContractTypeRow), "IdType", "IdContract")]
        public List<Int32> Contratcs
        {
            get { return Fields.Contracts[this]; }
            set { Fields.Contracts[this] = value; }
        }

doing that I have a multuiple select field in my **TypeForm** which allows me to add **contracts** to the **Type** (and insert in the table **type_contract** values (id_type, id_contract)

But when it comes to delete a contract from the multiple select field I receive this exception in the update method of TypeEndpoint
**Query affected 3 rows while 1 expected! This might mean that your query lacks a proper WHERE statement or a TRIGGER changes number of affected rows. In the latter case, you may try adding "SET NOCOUNT ON" to your trigger code.**

I found many solutions on github issues but all of them mention setting no count on on SQL server, I can't do that in the server, I don't know if there is a methode to override to change that

Thank you so much

**Comment by wezmag** [comment, marked as answer]
Please show us your `ContractTypeRow.cs`.
Based on your description, I think your `ContractType` table is missing an identity column.

**Reply by zbouchlih** [reply]
Thank you for your answer @wezmag 
Should I always have an identity attribute even if it's a many to many table ?

Here is my ContractTypeRow.cs

`using Serenity;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;
using System.IO;

namespace Tool
{
    [ConnectionKey("Default"), Module("Tool"), TableName("[dbo].[contract_type]")]
    [DisplayName("Contract Type"), InstanceName("Contract Type")]
    [ReadPermission(PermissionKeys.Contract.View)]
    [ModifyPermission(PermissionKeys.Contract.Modify)]
    [InsertPermission(PermissionKeys.Contract.Insert)]
    [DeletePermission(PermissionKeys.ContractType.Delete)]
    public sealed class ContractTypeRow : Row<ContractTypeRow.RowFields>, IIdRow
    {
        [DisplayName("Id Contract"), Column("id_contract"), PrimaryKey, ForeignKey("[dbo].[contract]", "id_contract"), LeftJoin("jIdContract"), IdProperty, TextualField("IdContractName")]
        public Int32? IdContract
        {
            get => fields.IdContract[this];
            set => fields.IdContract[this] = value;
        }
        [DisplayName("Id Type"), Column("id_type"), PrimaryKey, ForeignKey("[dbo].[type]", "id_type"), LeftJoin("jIdType"), TextualField("IdTypeName")]
        [LookupEditor(typeof(TypeRow))]
        public Int32? IdType
        {
            get => fields.IdType[this];
            set => fields.IdType[this] = value;
        }

        [Hidden, DisplayName("Date Creation"), Column("date_creation")]
        public DateTime? DateCreation
        {
            get => fields.DateCreation[this];
            set => fields.DateCreation[this] = value;
        }

        [DisplayName("Id Contract Name"), Expression("jIdContract.[name]")]
        public String IdContractName
        {
            get => fields.IdContractName[this];
            set => fields.IdContractName[this] = value;
        }

        [DisplayName("Id Type Name"), Expression("jIdType.[name]"), MinSelectLevel(SelectLevel.List)]
        public String IdTypeName
        {
            get => fields.IdTypeName[this];
            set => fields.IdTypeName[this] = value;
        }

        public ContractTypeRow()
            : base()
        {
        }

        public ContractTypeRow(RowFields fields)
            : base(fields)
        {
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field IdContract;
            public Int32Field IdType;
            public DateTimeField DateCreation;
            public StringField IdContractName;
            public StringField IdTypeName;

        }
    }
}`

**Reply by wezmag** [reply]
From my previous experience, in Serenity platform, an identity column is required in many-to-many intermediate table.
It does not have to be primary key, but it will be used for delete.
Check out source code of [LinkingSetRelationBehavior](https://github.com/serenity-is/Serenity/blob/523c0195fec17089e1dd7e812650c2edaf47a7ac/src/Serenity.Net.Services/RequestHandlers/IntegratedFeatures/LinkingSetRelation/LinkingSetRelationBehavior.cs#L420)

Actually, when using Serenity, it's better to have identity column on every table.

The movie database tutorial also have an identity column in MovieGenres table. 
https://serenity.is/docs/tutorials/movies/allowing_multiple_genre_selection

BTW, don't forget to move `IdProperty` attribute to the newly created identity column.

**Reply by zbouchlih** [reply]
Thank you so much for your help @wezmag  I added Identity column to my Many to Many table and it fixed the problem
